src/m2n/PointToPointCommunication.cpp

```diff
@@ -387,6 +389,74 @@ void PointToPointCommunication::acceptConnection(std::string const &acceptorName
   _isConnected = true;
 }
 
+void PointToPointCommunication::acceptPreConnection(std::string const &acceptorName,
+                                                 std::string const &requesterName)
+{
+  TRACE(acceptorName, requesterName);
+  assertion(not isConnected(), "Already connected!");
+  CHECK(utils::MasterSlave::isMaster() || utils::MasterSlave::isSlave(),
+        "You can only use a point-to-point communication between two participants which both use a master. "
+            << "Please use distribution-type gather-scatter instead.");
+
+  if (utils::MasterSlave::isMaster()) {
+    // Establish connection between participants' master processes.
+    auto c = _communicationFactory->newCommunication();
```

Comment:
Do we actually need this master-master connection? Nothing is exchanged here, right?

Reply:
Discussed with @atanasoa on the phone: let's remove it here and try to generate the data structure for QN not from the vertex distribution, but from the local information (in WP 6).

Reply:
ok, I agree. 
removed!